copier/renderer.py

```diff
@@ -0,0 +1,174 @@
+                / f"{result}{self.template.templates_suffix}"
+            )
+            if templated_sibling.exists():
+                return None
+        return result
+
+    def _render_file(self, src_abspath: Path) -> None:
+        """Render one file.
+
+        Args:
+            src_abspath:
+                The absolute path to the file that will be rendered.
+        """
+        # TODO Get from main.render_file()
+        assert src_abspath.is_absolute()
+        src_relpath = src_abspath.relative_to(self.template.local_abspath).as_posix()
+        src_renderpath = src_abspath.relative_to(self.template_copy_root)
+        dst_relpath = self._render_path(src_renderpath)
+        if dst_relpath is None:
+            return
+        if src_abspath.name.endswith(self.template.templates_suffix):
+            try:
+                tpl = self.jinja_env.get_template(src_relpath)
+            except UnicodeDecodeError:
+                if self.template.templates_suffix:
+                    # suffix is not empty, re-raise
+                    raise
+                # suffix is empty, fallback to copy
+                new_content = src_abspath.read_bytes()
+            else:
+                new_content = tpl.render(**self._render_context).encode()
+        else:
+            new_content = src_abspath.read_bytes()
+        dst_abspath = Path(self.subproject.local_abspath, dst_relpath)
+        src_mode = src_abspath.stat().st_mode
+        if not self._render_allowed(dst_relpath, expected_contents=new_content):
+            return
+        if not self.pretend:
+            dst_abspath.parent.mkdir(parents=True, exist_ok=True)
+            dst_abspath.write_bytes(new_content)
+            dst_abspath.chmod(src_mode)
+
+    def render_string(self, string: str) -> str:
+        """Render one templated string.
+
+        Args:
+            string:
+                The template source string.
+        """
+        tpl = self.jinja_env.from_string(string)
+        return tpl.render(**self._render_context)
+
+    @cached_property
+    def template_copy_root(self) -> Path:
+        """Absolute path from where to start copying.
+
+        It points to the cloned template local abspath + the rendered subdir, if any.
+        """
+        parts = [
+            self.template.local_abspath,
+        ]
+        subdir = self.render_string(self.template.subdirectory)
+        if subdir:
+            parts.append(subdir)
+
+        if self._items_key:
+            parts.append(self._items_key)
+
+        return Path(*parts)
+
+    def with_item(self, items_key: str, items_value: Any) -> "Renderer":
+        return Renderer(
+            self.template,
+            self.subproject,
+            self._render_allowed,
+            self.pretend,
+            self.jinja_env,
+            self.answers_relpath,
+            ChainMap({"_item": items_value}, self._render_context),
+            items_key,
+            [],
+        )
```

Comment:
You can do this probably:

```suggestion
        return replace(
            self,
            _render_context=ChainMap({"_item": items_value}, self._render_context),
            _items_key=items_key,
            _items_keys=[],
        )
```